tests/system/Cache/Handlers/RedisHandlerTest.php

```diff
@@ -129,44 +131,55 @@ public function testDelete(): void
         $this->assertFalse($this->handler->delete(self::$dummy));
     }
 
-    public function testDeleteMatchingPrefix(): void
+    #[DataProvider('provideDeleteMatching')]
+    public function testDeleteMatching(string $pattern, int $expectedDeleteCount, string $prefix = ''): void
     {
-        // Save 101 items to match on
+        $cache = new Cache();
+
+        if ($prefix !== '') {
+            $cache->prefix = $prefix;
+        }
+
+        /** @var RedisHandler $handler */
+        $handler = CacheFactory::getHandler($cache, 'redis');
+
         for ($i = 1; $i <= 101; $i++) {
-            $this->handler->save('key_' . $i, 'value' . $i);
+            $handler->save('key_' . $i, 'value_' . $i);
         }
 
-        // check that there are 101 items is cache store
-        $dbInfo = explode(',', $this->handler->getCacheInfo()['db0']);
-        $this->assertSame('keys=101', $dbInfo[0]);
+        $cacheInfo = $handler->getCacheInfo();
+        $this->assertIsArray($cacheInfo);
+        $this->assertArrayHasKey('db0', $cacheInfo);
+        $this->assertIsString($cacheInfo['db0']);
+        $this->assertMatchesRegularExpression('/^keys=(?P<count>\d+)/', $cacheInfo['db0']);
+
+        preg_match('/^keys=(?P<count>\d+)/', $cacheInfo['db0'], $matches);
+        $this->assertSame(101, (int) $matches['count']);
+
+        $this->assertSame($expectedDeleteCount, $handler->deleteMatching($pattern));
 
-        // Checking that given the prefix "key_1", deleteMatching deletes 13 keys:
-        // (key_1, key_10, key_11, key_12, key_13, key_14, key_15, key_16, key_17, key_18, key_19, key_100, key_101)
-        $this->assertSame(13, $this->handler->deleteMatching('key_1*'));
+        $cacheInfo = $handler->getCacheInfo();
+        $this->assertIsArray($cacheInfo);
+        $this->assertArrayHasKey('db0', $cacheInfo);
+        $this->assertIsString($cacheInfo['db0']);
+        $this->assertMatchesRegularExpression('/^keys=(?P<count>\d+)/', $cacheInfo['db0']);
 
-        // check that there remains (101 - 13) = 88 items is cache store
-        $dbInfo = explode(',', $this->handler->getCacheInfo()['db0']);
-        $this->assertSame('keys=88', $dbInfo[0]);
+        preg_match('/^keys=(?P<count>\d+)/', $cacheInfo['db0'], $matches);
+        $this->assertSame(101 - $expectedDeleteCount, (int) $matches['count']);
+
+        $handler->deleteMatching('key_*');
     }
 
-    public function testDeleteMatchingSuffix(): void
+    /**
+     * @return iterable<string, array{0: string, 1: int, 2?: string}>
+     */
+    public static function provideDeleteMatching(): iterable
     {
-        // Save 101 items to match on
-        for ($i = 1; $i <= 101; $i++) {
-            $this->handler->save('key_' . $i, 'value' . $i);
-        }
-
-        // check that there are 101 items is cache store
-        $dbInfo = explode(',', $this->handler->getCacheInfo()['db0']);
-        $this->assertSame('keys=101', $dbInfo[0]);
+        yield 'prefix' => ['key_1*', 13];
```

Comment:
Can you add the comment why the count is 13?

Reply:
```suggestion
        // Given the key "key_1*", deleteMatching() should delete 13 keys:
        // key_1, key_10, key_11, key_12, key_13, key_14, key_15, key_16, key_17, key_18, key_19, key_100, key_101
        yield 'prefix' => ['key_1*', 13];
```